core/vector/vector_types.rs

```diff
@@ -316,6 +316,71 @@ pub fn vector_type(blob: &[u8]) -> Result<VectorType> {
     }
 }
 
+pub fn vector_concat(v1: &Vector, v2: &Vector) -> Result<Vector> {
+    if v1.vector_type != v2.vector_type {
+        return Err(LimboError::ConversionError(
+            "Mismatched vector types".into(),
+        ));
+    }
+
+    let mut data = Vec::with_capacity(v1.data.len() + v2.data.len());
+    data.extend_from_slice(&v1.data);
+    data.extend_from_slice(&v2.data);
+
+    Ok(Vector {
+        vector_type: v1.vector_type,
+        dims: v1.dims + v2.dims,
+        data,
+    })
+}
+
+pub fn vector_slice(vector: &Vector, start_idx: usize, end_idx: usize) -> Result<Vector> {
+    fn extract_bytes<T, const N: usize>(
+        slice: &[T],
+        start: usize,
+        end: usize,
+        to_bytes: impl Fn(&T) -> [u8; N],
+    ) -> Result<Vec<u8>> {
+        if start > end {
+            return Err(LimboError::InvalidArgument(
+                "start index must not be greater than end index".into(),
+            ));
+        }
+        if end > slice.len() || start >= slice.len() {
```

Comment:
NIT: I think we can allow empty slice at the end like `let x = [1, 2, 3]; let y = &x[3..3]`. Just because it make some sense actually (so you can drop `start >= slice.len()` condition as you already validated `end` and `start <= end`)